### PR TITLE
add 'predeclared' linter

### PR DESCRIPTION
**Pull request by nishanths**
https://github.com/nishanths/predeclared

This adds the `predeclared` linter, which reports code that shadows one of [Go's predeclared identifiers](https://golang.org/ref/spec#Predeclared_identifiers).

### CODE REVIEW

**Comment by nishanths** [comment]
Fixed the tests. This should be ready for review!

**Comment by nishanths** [comment]
Fixed the tests. This should be ready for review!

**Comment by nishanths** [comment]
Fixed the tests. This should be ready for review.

**Review by iwankgb — APPROVED** [review]

**Comment by iwankgb** [comment]
@golangci/team is anyone opposed to adding this linter?

**Comment by Nivl** [comment]
> @golangci/team is anyone opposed to adding this linter?

I'm not. I actually like it a lot.

**Comment by magnetikonline** [comment]
Yep, agree too. TIL this was actually possible and this is valid Go code:

```go
package main

import "fmt"

func main() {
	fmt.Println("Hello")
	append()
}

func append() {
	fmt.Println("Hello")
}
```

**Comment by nishanths** [comment]
Thanks for reviewing. It looks like I have the ability to merge this pull request. Shall I go ahead, or should I wait for a maintainer to merge this/more reviews?

**Pull request merged**
